language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java

```diff
@@ -134,6 +135,22 @@ public static CodeActionNodeType topLevelNodeInLine(LSContext context, TextDocum
                         && cursorLine == diagnosticPos.sLine) {
                     return CodeActionNodeType.RECORD;
                 }
+
+                if (topLevelNode instanceof BLangClassDefinition) {
```

Comment:
Since we still use the AST nodes to get the node type, let's skip using the BLangNode. Let's calculate the top level node type to be the Syntax node type directly.

Reply:
Yes agree, only added changes affected by add-doc code action changes.  Let's remove it coming up PRs.

Reply:
+1